### PR TITLE
OHRM5X-1751: Develop Dashboard Employee On Leave Today Configuration API

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmDashboardPlugin/config/routes.yaml [hunk 1]
```diff
@@ -42,6 +42,14 @@ apiv2_dashboard_employee_action_summary:
     _api: OrangeHRM\Dashboard\Api\EmployeeActionSummaryAPI
     id: 0
 
+apiv2_dashboard_employee_on_leave_today_configuration:
+  path: /api/v2/dashboard/config/employee-on-leave-today
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/config/routes.yaml:46]:
```suggestion
apiv2_dashboard_employees_on_leave_today_configuration:
  path: /api/v2/dashboard/config/employees-on-leave-today
```
:thinking:

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Service\ConfigServiceTrait;
+use OrangeHRM\Dashboard\Service\EmployeeOnLeaveService;
+
+class EmployeeOnLeaveTodayConfigAPI extends Endpoint implements ResourceEndpoint
+{
+    use ConfigServiceTrait;
+
+    public const PARAMETER_CONFIG_ENABLED = 'isEnabled';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:40]: *

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    public function update(): EndpointResult
+    {
+        $isEnabled = $this->getRequestParams()->getBoolean(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_CONFIG_ENABLED
+        );
+
+        $config = $this->getConfigService()
+            ->getConfigDao()
+            ->setValue(EmployeeOnLeaveService::CONFIG_ONLY_SHOW_EMPLOYEES_REPORTING_TO_ME, (int)$isEnabled);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:75]: *

File: installer/Migration/V5_2_0/permission/api.yaml [hunk 4]
```diff
@@ -77,6 +77,18 @@ apiv2_dashboard_employee_action_summary:
     - { role: Admin, permission: { read: true, create: false, update: false, delete: false, self: false } }
     - { role: ESS, permission: { read: true, create: false, update: false, delete: false, self: false } }
 
+apiv2_dashboard_employee_on_leave_today_configuration:
+  description: "API-v2 Dashboard - Employee On Leave Today Configuration"
```
Comment (RajithaKumara) [inline installer/Migration/V5_2_0/permission/api.yaml:81]:
```suggestion
apiv2_dashboard_employees_on_leave_today_configuration:
  description: "API-v2 Dashboard - Employees On Leave Today Configuration"
```
:thinking:

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
We may have to add new datagroup to conditionally show the setting weel only for Admin users.

File: src/plugins/orangehrmCorePlugin/Service/ConfigService.php [hunk 5]
```diff
@@ -51,6 +51,7 @@ class ConfigService
     public const KEY_INSTANCE_IDENTIFIER_CHECKSUM = 'instance.identifier_checksum';
     public const KEY_SENDMAIL_PATH = 'email_config.sendmail_path';
     public const KEY_LDAP_SETTINGS = 'ldap_settings';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Service/ConfigService.php:53]:
```suggestion
    public const KEY_LDAP_SETTINGS = 'admin.ldap_settings';
```
:thinking: what do you think? Seems this is my mistake

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveAPI.php [hunk 6]
```diff
@@ -62,6 +69,14 @@ public function getAll(): EndpointResult
 
         $employeeOnLeaveSearchFilterParams->setDate($date);
 
+        $showOnlyAccessibleEmployeesOnLeaveToday = $this->getConfigService()
+            ->getDashboardEmployeesOnLeaveTodayShowOnlyAccessibleConfig();
+
+        if ($showOnlyAccessibleEmployeesOnLeaveToday) {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $employeeOnLeaveSearchFilterParams->setAccessibleEmpNumber([$this->getAuthUser()->getEmpNumber(),...$accessibleEmpNumbers]);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveAPI.php:77]:
```php
            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
            $accessibleEmpNumbers[] = $this->getAuthUser()->getEmpNumber();
            $employeeOnLeaveSearchFilterParams->setAccessibleEmpNumber($accessibleEmpNumbers);
```
this also same

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $showOnlyAccessibleEmployeesOnLeaveToday = $this->getConfigService()
+            ->getDashboardEmployeesOnLeaveTodayShowOnlyAccessibleConfig();
+        return new EndpointResourceResult(
+            ArrayModel::class,
+            ['showOnlyAccessibleEmployeesOnLeaveToday' => $showOnlyAccessibleEmployeesOnLeaveToday]
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:50]:
```suggestion
            [self::PARAMETER_SHOW_ONLY_ACCESSIBLE_EMPLOYEES_ON_LEAVE_TODAY=> $showOnlyAccessibleEmployeesOnLeaveToday]
```

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            self::PARAMETER_SHOW_ONLY_ACCESSIBLE_EMPLOYEES_ON_LEAVE_TODAY
+        );
+
+        $this->getConfigService()->setDashboardEmployeesOnLeaveTodayShowOnlyAccessibleConfig(
+            $showOnlyAccessibleEmployeesOnLeaveToday
+        );
+
+        return new EndpointResourceResult(
+            ArrayModel::class,
+            ['showOnlyAccessibleEmployeesOnLeaveToday' => $showOnlyAccessibleEmployeesOnLeaveToday]
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:80]:
```suggestion
            [self::PARAMETER_SHOW_ONLY_ACCESSIBLE_EMPLOYEES_ON_LEAVE_TODAY=> $showOnlyAccessibleEmployeesOnLeaveToday]
```

File: src/plugins/orangehrmDashboardPlugin/Dto/EmployeeOnLeaveSearchFilterParams.php [hunk 9]
```diff
@@ -49,4 +57,36 @@ public function setDate(DateTime $dateTime): void
...
+        return $this->dateTime;
+    }
+
+    /**
+     * @param DateTime $dateTime
+     */
+    public function setDateTime(DateTime $dateTime): void
+    {
+        $this->dateTime = $dateTime;
+    }
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Dto/EmployeeOnLeaveSearchFilterParams.php:75]:
```suggestion
```
remove these getter and setter, `getDate`, `setDate` already there

**Review by RajithaKumara — APPROVED** [review]
check code coverage

File: src/plugins/orangehrmDashboardPlugin/Dto/EmployeeOnLeaveSearchFilterParams.php (hunk 9 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Dto/EmployeeOnLeaveSearchFilterParams.php:75]: Oops, this was my mistake when generating the getters and setters

File: src/plugins/orangehrmCorePlugin/Service/ConfigService.php (hunk 5 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Service/ConfigService.php:53]: yeah saw this one and wondered why there is no prefix. Shall I add the prefix

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php (hunk 7 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:50]: done

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php (hunk 8 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveTodayConfigAPI.php:80]: done

**Pull request merged**